arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/raise/RaiseAccumulate.kt

```diff
@@ -0,0 +1,570 @@
+@file:OptIn(ExperimentalTypeInference::class, ExperimentalContracts::class)
+@file:JvmMultifileClass
+@file:JvmName("RaiseKt")
+package arrow.core.raise
+
+import arrow.core.Either
+import arrow.core.EmptyValue
+import arrow.core.NonEmptyList
+import arrow.core.Tuple4
+import arrow.core.Tuple5
+import arrow.core.Tuple6
+import arrow.core.Tuple7
+import arrow.core.Tuple8
+import arrow.core.emptyCombine
+import arrow.core.nel
+import arrow.typeclasses.Semigroup
+import kotlin.contracts.ExperimentalContracts
+import kotlin.contracts.InvocationKind
+import kotlin.contracts.contract
+import kotlin.experimental.ExperimentalTypeInference
+import kotlin.jvm.JvmMultifileClass
+import kotlin.jvm.JvmName
+
+
+/**
+ * Accumulate the errors from running both [action1] and [action2]
+ * using the given [semigroup].
+ */
+@RaiseDSL
+public inline fun <R, A, B, C> Raise<R>.zipOrAccumulate(
+  semigroup: Semigroup<@UnsafeVariance R>,
+  @BuilderInference action1: Raise<R>.() -> A,
+  @BuilderInference action2: Raise<R>.() -> B,
+  @BuilderInference block: Raise<R>.(A, B) -> C
+): C {
+  contract {
+    callsInPlace(action1, InvocationKind.EXACTLY_ONCE)
+    callsInPlace(action2, InvocationKind.EXACTLY_ONCE)
+  }
```

Comment:
Right, now we can also add `contract` 🥳

Reply:
I think 50% of the code starts with `callsInPlace` xD